src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service
+
+import mate.academy.model.Circle
+import mate.academy.model.Color
+import mate.academy.model.IsoscelesTrapezoid
+import mate.academy.model.Rectangle
+import mate.academy.model.RightTriangle
+import mate.academy.model.Square
+import kotlin.random.Random
+
+const val MIN_RANDOM_SIZE = 1.0
+const val MAX_RANDOM_SIZE = 10.0
+
+class FigureSupplier {
+
+    private val colorSupplier = ColorSupplier()
+
+    private fun getRandomFigureType(): FigureType {
+        return FigureType.values().random()
+    }
+
+    fun getRandomFigure(): Figure {
+        return when (getRandomFigureType()) {
+            FigureType.SQUARE -> Square(colorSupplier.getRandomColor(),
```

Comment:
Note, that FigureType could be imported to improve readability. Just press ALT+ENTER when cursor on required item and select this option